### PR TITLE
fix(YouTube - Player flyout menu): Restore functionality

### PR DESCRIPTION
**Pull request by johnconner122**
depends on:

 - ReVanced/revanced-patches/pull/3163
 

image

### CODE REVIEW

**Comment by oSumAtrIX** [comment]
What about the other list entries?

**Comment by johnconner122** [comment]
> What about the other list entries?

They are shown after clicking advance menu:
Screenshot_2023-10-18-14-22-00-831_app revanced android youtube

**Comment by johnconner122** [comment]
> > What about the other list entries?
> 
> They are shown after clicking additional settings menu: Screenshot_2023-10-18-14-22-00-831_app revanced android youtube

**Comment by johnconner122** [comment]
`yt_outline_gear` hides `quality` menu in old flyout but in new flyout it hides `additional settings` menu. I think best is to add `quality` menu to exceptions(Quality menu can't be hidden) and hide `additional settings` menu only.

**Comment by oSumAtrIX** [comment]
We can remove hiding essential list items such as quality or speed. Usually, the useless items include VR & "Help and feedback" so reducing the patch down to that would be good enough.

File: app/src/main/java/app/revanced/integrations/patches/components/PlayerFlyoutMenuItemsFilter.java [hunk 1]
```diff
@@ -6,26 +6,36 @@
...
     @RequiresApi(api = Build.VERSION_CODES.N)
     public PlayerFlyoutMenuItemsFilter() {
-        identifierFilterGroupList.addAll(new StringFilterGroup(null, "overflow_menu_item.eml|"));
+        exceptions.addAll(
+                new ByteArrayAsStringFilterGroup(null, "quality_sheet") // Whitelist Quality menu item
+        );
+
+        // Using pathFilterGroupList due to new flyout panel(A/B)
+        pathFilterGroupList.addAll(
+                new StringFilterGroup(null, "overflow_menu_item.eml|")
```
[the remark this answers is not in the document]
Comment (johnconner122) [reply app/src/main/java/app/revanced/integrations/patches/components/PlayerFlyoutMenuItemsFilter.java:27]: Should I check for `setting` when to search for `exception` or not, like:

```shell
exception = new ByteArrayAsStringFilterGroup(
                SettingsEnum.HIDE_ADDITIONAL_SETTINGS_MENU,
                "quality_sheet" // Whitelist Quality menu item when "Hide Additional settings menu" is enabled
        );
```

**Comment by oSumAtrIX** [comment]
@johnconner122 Could you reach out to me at osumatrix@revanced.app. You do not have any other public contact method that I can see.

**Comment by johnconner122** [comment]
> @johnconner122 Could you reach out to me at [osumatrix@revanced.app](mailto:osumatrix@revanced.app). You do not have any other public contact method that I can see.

I have sent you an email.

**Comment by johnconner122** [comment]
Should this menu divider be removed? It doesn't feel natural there:

 276256756-42e95a19-1afe-4aa4-9b93-c0fed82cfccc

**Review by oSumAtrIX — APPROVED** [review]

**Pull request merged**